test/pecos/utils/test_mmap_hashmap_util.py

```diff
@@ -66,6 +66,116 @@ def test_str2int_mmap_hashmap(tmpdir):
     assert r_map_batch_getter.get(ks, 10).tolist() == vs
 
 
+def test_fixed_len_str2int_mmap_hashmap(tmpdir):
+    from pecos.utils.mmap_hashmap_util import MmapHashmap, MmapHashmapBatchGetter
+
+    map_dir = tmpdir.join("fixed_len_str2int").realpath().strpath
+    kv_dict = {"aaa".encode("utf-8"): 2, "bbb".encode("utf-8"): 3}
+
+    # Write-only Mode
+    w_map = MmapHashmap("fixed_len_str2int")
+    w_map.open("w", map_dir)
+    # Insert
+    w_map.map.insert("aaa".encode("utf-8"), 1)  # Test for overwrite later
+    for k, v in kv_dict.items():
+        w_map.map.insert(k, v)
+    # Size
+    assert w_map.map.size() == len(kv_dict)
+    w_map.close()
+
+    # Read-only Mode
+    r_map = MmapHashmap("fixed_len_str2int")
+    r_map.open("r", map_dir)
+    # Get
+    for k, v in kv_dict.items():
+        assert r_map.map[k] == v
+    # Get with default
+    for k, v in kv_dict.items():
+        assert r_map.map.get(k, 10) == v
+    assert r_map.map.get("ccc".encode("utf-8"), 10) == 10
+    # Contains
+    for k, _ in kv_dict.items():
+        assert k in r_map.map
+    assert not ("ccc".encode("utf-8") in r_map.map)
+    # Size
+    assert r_map.map.size() == len(kv_dict)
+
+    # Batch get with default
+    max_batch_size = 5
+    # max_batch_size > num of key
+    r_map_batch_getter = MmapHashmapBatchGetter(r_map.map, max_batch_size)
+    ks = list(kv_dict.keys()) + ["ccc".encode("utf-8")]  # Non-exist key
+    vs = list(kv_dict.values()) + [10]
+    assert r_map_batch_getter.get(ks, 10).tolist() == vs
+    # max_batch_size = num of key
+    ks = list(kv_dict.keys()) + ["ccc".encode("utf-8")] * (
+        max_batch_size - len(kv_dict)
+    )  # Non-exist key
+    vs = list(kv_dict.values()) + [10] * (max_batch_size - len(kv_dict))
+    assert r_map_batch_getter.get(ks, 10).tolist() == vs
```

Comment:
Are you testing increasing `max_batch_size` inside class? Then you should check it indeed got increased.